### DISCUSSION TITLE
Make languages pub structs and place them into corresonding Lang enum variants

### DISCUSSION THREAD

**Discussion by pavloslav** [Ballasi/num2words#22] [Ideas]
Now, users only get common Num2Words interface, which is simpler, but involves more operations, like string comparisons and Boxing. Many rustaceans will hate it.

Also, let's derive all common traits on Num2Words and (if public) languages, like PartialEq, Clone, Hash, to allow using them, say, in HashMaps.

**Comment by Laifsyn** [comment]
I would like to add on this. I dislike how I constantly have to convert from the Num2Word::new().

This new() method should've been relieved to the Default trait impl instead of to `Num2Words::new()` 
Even if ergonomic, it's been a pain for Integration Test to always have to call methods to edit the default language.

I would suggest that  Num2Words should be treated as a "driver" instead of a disposable struct since Num2Words weights 56 Bytes (not taking into account the space that might be needed from the heap for each instance of String it holds in preferences.
